packages/amplify-e2e-tests/src/__tests__/mock/auth-transformer-mock.test.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  addApiWithAllAuthModes,
+  amplifyMock,
+  createNewProjectDir,
+  deleteProject,
+  deleteProjectDir,
+  GraphQLClient,
+  initJSProjectWithProfile,
+  updateApiSchema,
+} from '@aws-amplify/amplify-e2e-core';
+
+describe('Test auth transformer with mock', () => {
+  const projName = 'authmock';
+  const apiKey = 'da-fake-api-key';
+  const endpoint = 'http://localhost:20002';
+  let projRoot: string;
+  let executionContext;
+  let graphqlClient;
+  beforeEach(async () => {
+    projRoot = await createNewProjectDir(projName);
+    await initJSProjectWithProfile(projRoot, { name: projName });
+    await addApiWithAllAuthModes(projRoot, { transformerVersion: 2 });
+    await updateApiSchema(projRoot, projName, 'simple_model_api_key.graphql');
+    executionContext = amplifyMock(projRoot);
+    graphqlClient = new GraphQLClient(endpoint, { 'x-api-key': apiKey });
+  });
+
+  afterEach(async () => {
+    await executionContext.sendCtrlC().runAsync();
+    await deleteProject(projRoot);
+    deleteProjectDir(projRoot);
```

Comment:
nit: extra line same as above. Do we need both of them (async & sync delete) ?
Or did you mean to do something like:
```

      const mockFilePath = path.join(projRoot, 'amplify', 'mock-data');
      if (existsSync(metaFilePath)) {
        await deleteProject(projRoot);
      }
      deleteProjectDir(projRoot);
     
```